### ISSUE TITLE
HPA uses wrong count to calculate target replcias

### ISSUE THREAD

**Issue by mwielgus** [kubernetes/kubernetes#34821]
The current formula looks like: 

``` go
currentReplicas := scale.Status.Replicas
[...] 
usageRatio := float64(utilization) / float64(targetUtilization)
    if math.Abs(1.0-usageRatio) > tolerance {
        return int32(math.Ceil(usageRatio * float64(currentReplicas))), &utilization, timestamp, nil
    }
```

If currentReplicas doesn't match to the number of replicas used to calculate utilization then the result can be way bigger than necessary. For example if there are 10 pods created in apiserver (= current replicas) but only 1 is scheduled and it is running at its last legs (300% of the target utilization) then the target will grow to 30 pods.
20 (30-10) pods get created and are not scheduled and will increase the replicas even further.

Solly you did some fixes in this area: https://github.com/kubernetes/kubernetes/pull/33593. 
Will this PR fix the above scenario?

cc: @DirectXMan12  @jszczepkowski @fgrzadkowski @davidopp

**Comment by jszczepkowski** [comment]
If this is urgent and we need to fix it on 1.4, I can prepare a small, separate PR with the fix. PR #33593 is huge and will not be cherry-picked to 1.4.

**Comment by DirectXMan12** [comment]
#33593 currently experiences this as well, I think, but I'm trying to figure out the right way to solve it.  The problem, as I see it, is this:  while we can be reasonably certain that an unready pod will probably become ready eventually (certain-ish, at least), a pod in a pending state may effectively be permafailed (e.g. if you don't specify a command, it looks like the pod will report pending, unready, with one of the containers reporting that the runtime indicated that no command was specified).  So, we need to figure out what we're going to guess about pending pods.  I need to do a bit more digging about pod lifecycle.

**Comment by davidopp** [comment]
cc/ @lavalamp for the HPA problem we saw today

**Comment by derekwaynecarr** [comment]
@DirectXMan12 and I chatted on this today, I think we should split pods into two states based on the pod condition `Ready`.  I would ignore cpu usage associated with pods whose `Ready` condition was not true.

**Comment by jayunit100** [comment]
is this sorted now ?

**Comment by DirectXMan12** [comment]
it should be fixed as part of #33593

**Comment by jszczepkowski** [comment]
Fixed by #33593.
